### DISCUSSION TITLE
Handling request.json() Without Async in Starlette

### DISCUSSION THREAD

**Discussion by yurifeigin** [encode/starlette#2504] [Q&A] [answered]
Hi everyone,

I encountered an interesting challenge while working with Starlette. I have a `def` function (not `async`) that handles HTTP POST requests. However, I need to extract JSON data from the request body using `request.json()`. Since I can't use `await` in a regular `def` function, I tried using `asyncio.run(request.json())`, but it occasionally throws exceptions like:
`RuntimeError: Task <Task pending name=‘Task-1639’ coro=<Request.json() running at lib/python3.9/site-packages/starlette/requests.py:249> cb=[_run_until_complete_cb() at lib/python3.9/asyncio/base_events.py:184]> got Future <Future pending> attached to a different loop`

What would be the correct approach to handle `request.json()` without using `async`? I'd appreciate any insights or alternative solutions.

Thanks in advance!

**Comment by Kludex** [comment, marked as answer]
You need to use `async`. Run the rest in a thread executor.
